students/MichaelM/lesson5/mailroom_rev_try_catch.py

```diff
@@ -0,0 +1,258 @@
+import os
+import pathlib
+
+pth = pathlib.Path('./')
+tmp_directory = "{}/tmp/".format(os.getcwd())
+
+esteemed_donors_headers = ["Donor Name", "Total Donation", "Number of Donations", "Ave Donation"]
+esteemed_donors = [["John Jacob Astor", 10.0],
+                   ["John Jacob Astor", 100.0],
+                   ["John Jacob Astor", 1000.0],
+                   ["John Jacob Astor", 10000.0],
+                   ["Alan Rufus, 1st Lord of Richmond", 10.0],
+                   ["Henry Ford", 10.0],
+                   ["Henry Ford", 100.0],
+                   ["Henry Ford", 1000.0],
+                   ["Cornelius Vanderbilt", 10.0],
+                   ["Jakob Fugger", 10.0],
+                   ["Jakob Fugger", 100.0]]
```

Comment:
at this point you should be using a dictionary data structure here which was also called out in the previous submission.